Algorithm/QCAlgorithm.cs

```diff
@@ -1458,9 +1474,9 @@ public Option AddOption(string underlying, Resolution resolution = Resolution.Mi
             }
 
             var marketHoursEntry = MarketHoursDatabase.GetEntry(market, underlying, SecurityType.Option);
-            var symbolProperties = _symbolPropertiesDatabase.GetSymbolProperties(market, underlying, SecurityType.Option, CashBook.AccountCurrency);
+            var symbolProperties = _symbolPropertiesDatabase.GetSymbolProperties(market, underlying, SecurityType.Option, Portfolio.CashBook.AccountCurrency);
             var canonicalSecurity = (Option) SecurityManager.CreateSecurity(typeof(ZipEntryName), Portfolio, SubscriptionManager,
-                marketHoursEntry.ExchangeHours, marketHoursEntry.DataTimeZone, symbolProperties, SecurityInitializer, canonicalSymbol, resolution,
+                marketHoursEntry.ExchangeHours, marketHoursEntry.DataTimeZone, symbolProperties, SecurityInitializer, canonicalSymbol, resolution, Portfolio.CashBook.AccountCurrency,
```

Comment:
This method already accepts the `Portoflio` object, so no need to pass the account currency explicitly. This comment applies to many other places in this PR as well.